webapp/src/pages/ResponsePage/ResponsePage.js

```diff
@@ -106,6 +110,162 @@ class ResponsePage extends Component {
         };
     };
 
+    getOutcome() {
+        if (this.state.checkOutcome) {
+            outcomeService.getOutcome(this.props.event.id, this.state.applicationData.user_id).then(response => {
+                if (response.status === 200) {
+                    const newOutcome = {
+                        timestamp: response.outcome.timestamp,
+                        status: response.outcome.status,
+                    };
+                    this.setState(
+                        {
+                            outcome: newOutcome
+                        }
+                    );
+                }
+                else{
+                    this.error(response.error)
+                }
+                this.setState(
+                    {
+                        checkOutcome: false,
+                    }
+                )
+                
+            });
+        }
+    };
+
+    submitOutcome(selectedOutcome) {
+        outcomeService.assignOutcome(this.state.applicationData.user_id, this.props.event.id, selectedOutcome).then(response => {
+            if (response.status === 201) {
+                const newOutcome = {
+                    timestamp: response.outcome.timestamp,
+                    status: response.outcome.status,
+                };
+
+                this.setState({
+                    outcome: newOutcome
+                });
+            } else {
+                this.error(response.error);
+            }
+        });
+    }
+
+    outcomeStatus() {
+        const data = this.state.applicationData;
+        
+        if (data) {
+
+            this.getOutcome();
```

Comment:
Note that setState calls in React are aync, so there's no guarantee that the code you have here after calling `getOutcome` will run before or after `getOutcome` actually updates the outcome in `this.state`. In order to ensure that code which depends on some state being set runs only after that state has been set, you need to use the callback in `setState`. (See [this explanation](https://stackoverflow.com/a/42038724) on StackOverflow for more info)

Reply:
Also from the chain of calls it looks like getOutcome is effectively being called inside the render() function? that's not a good idea because render gets called every time state is updated, so you could get an infinite loop unless you do something like what you're doing with checkOutcome. But that should not be needed. Instead it's better to get the outcome in `componentDidMount` where all the other API calls are and then you won't need the checkOutcome either.